### PR TITLE
ENH: Add read_logfmt()

### LINKED ISSUE #34501: ENH: add logfmt support
[logfmt](https://www.brandur.org/logfmt) is a structured log format similar to JSON lines, but much simpler. Each line has a format `column1=value1 column2=value2 ...`. Pandas already has support for JSON lines, which can be read with `pd.read_json(path, lines=True)`, but to read logfmt logs a conversion is needed.

I have been using [agrind](https://github.com/rcoh/angle-grinder) to convert logs to JSON lines, but it is an additional step that I would like to eliminate from my workflow. You can see an example at https://gitlab.com/nsnam/ns-3-dev/-/merge_requests/303

Besides, I would like to use logfmt as an output format for simple computational experiments (think C/Fortran without dependencies), because it is easier to maintain one line of code that outputs keys and values interleaved, in contrast to maintaining the code to output headers and code to output values separately. And logfmt becomes easier to read than CSV as the number of columns grows. Compare
```
node=1 throughput=63.0 delay=52.0 x=0.0 y=5.0
node=2 throughput=100.0 delay=35.0 x=5.0 y=5.0
```
to
```
node,throughput,delay,x,y
1,63.0,52.0,0.0,5.0
2,100.0,35.0,5.0,5.0
```

To read logfmt directly with pandas, I want a function `pd.read_logfmt()` that can be used in place of `pd.read_csv()` and `pd.read_json()` in described above cases.

### PR DESCRIPTION
**Pull request by link2xt**
- [x] closes #34501 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by link2xt** [comment]
Doesn't pass CI yet becasue conda-forge does not have `logfmt` package.

File: pandas/io/logfmt_format.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                except ValueError:
+                    try:
+                        line[key] = float(line[key])
+                    except ValueError:
+                        pass
+            yield line
+
+    def __next__(self):
+        import_optional_dependency("logfmt")
+        import logfmt
```
Comment (MarcoGorelli) [inline pandas/io/logfmt_format.py:153]: Should this be
```suggestion
        logfmt = import_optional_dependency("logfmt")
```
?

Anyway, sorry if I'm missing some context, but is there an open issue about adding this to the API? Did you check that the core team would be open to having it?

**Review by jreback — CHANGES_REQUESTED** [review]
can u open an issue for discussion about this first

**Comment by link2xt** [comment]
> can u open an issue for discussion about this first

Did not know it was required, [contributing guide](https://pandas.pydata.org/docs/development/contributing.html) asks to search for issue, but "Bug reports and enhancement requests" section does not discuss enhancement requests. Here is an issue: https://github.com/pandas-dev/pandas/issues/34501

File: pandas/io/logfmt_format.py (hunk 1 above)
Comment (link2xt) [reply pandas/io/logfmt_format.py]: Changed this line, though `import logfmt` works too.

Opened an issue #34501 for enhancement request.

**Pull request closed without merging**